.circleci/config.yml

```diff
@@ -48,28 +48,33 @@ jobs:
           command: |
             curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | POETRY_UNINSTALL=1 python -
             curl -sSL https://install.python-poetry.org | python3 -
-      - python/install-packages:
-          pkg-manager: poetry
-          # args: "--extras docs"
-          cache-version: "v1" # change to clear cache
-      - run:
-          name: Check poetry package versions
-          command: |
-            poetry --version
-            poetry show
       - run:
           name: Set BASH_ENV
           command: |
             set -e
             sudo apt update
-            sudo apt install -qq graphviz optipng libxft2
+            sudo apt install -qq graphviz optipng libxft2 graphviz-dev
             echo "set -e" >> $BASH_ENV
             echo "export OPENBLAS_NUM_THREADS=4" >> $BASH_ENV
             echo "export XDG_RUNTIME_DIR=/tmp/runtime-circleci" >> $BASH_ENV
             echo "export PATH=~/.local/bin/:$PATH" >> $BASH_ENV
             echo "export DISPLAY=:99" >> $BASH_ENV
             echo "BASH_ENV:"
             cat $BASH_ENV
+      - run:
+          name: Install pysal dependencies
+          command: |
+            sudo apt install libspatialindex-dev xdg-utils
+      - python/install-packages:
+          pkg-manager: poetry
+          args: "--extras graph_func"
+          cache-version: "v1" # change to clear cache
+      - run:
+          name: Check poetry package versions
+          command: |
+            poetry run pip install --upgrade --progress-bar off pygraphviz
```

Comment:
This seems weird - why not add `pygraphviz` as a listed dependency in `pyproject.toml`? In general, my approach in dowhy has been to avoid `pip` entirely and model all dependencies inside of `pyproject.toml'. When we have these tacit installs it makes setting up a dev environment kind of tricky.

Reply:
Good point. I added it to the `pyproject.toml` and updated the poetry.lock file in #19